pontoon/base/views.py

```diff
@@ -158,9 +159,51 @@ def projects(request, template='projects.html'):
     return render(request, template, data)
 
 
-def translate(request, locale, slug, part=None, template='translate.html'):
+def locale_project(request, locale, slug):
+    """Locale-project overview."""
+
+    # Redirect to translate view
+    if request.GET.get('q'):
+        return translate(request, locale, slug, '')
+
+    try:
+        l = Locale.objects.get(code__iexact=locale)
+    except Locale.DoesNotExist:
+        raise Http404
+
+    # Prefetch subpages and resources for locales_parts_stats()
+    projects = (
+        Project.objects.filter(
+            disabled=False,
+            slug=slug,
+            pk__in=Resource.objects.values('project')
+        )
+        .prefetch_related(
```

Comment:
You've officially written this prefetch twice, time to refactor it to a queryset method!